lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -0,0 +1,344 @@
+    setElToFocusId('');
+    setFocusableBtnIndex(null);
+    setShowDropdown(false);
+  }
+
+  function handleElInView(el: HTMLElement) {
+    const bounding = el.getBoundingClientRect();
+    const dropdownMenu = dropdownRef.current;
+
+    //this check if the dropdown menu is not in the document view
+    const isInView = bounding.top >= 0 && bounding.left >= 0 &&
+      bounding.bottom <= (document.documentElement.clientHeight)
+      && bounding.right <= (document.documentElement.clientWidth);
+
+    if (!isInView) {
+      el.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
+    }
+
+    //this check if the option is not visible inside the dropdown menu
+    if (dropdownMenu &&
+      dropdownMenu.scrollTop + dropdownMenu.offsetHeight <
+      el.offsetTop + el.offsetHeight
+    ) {
+      el.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
+    }
+    else if (dropdownMenu && dropdownMenu.scrollTop > el.offsetTop + 4) {
+      el.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
+    }
+  }
+
+  return (
+    /*eslint-disable-next-line jsx-a11y/no-static-element-interactions*/
+    <div ref={ref}
+         className={className}
+         onFocus={handleComponentFocus}
+         onBlur={handleComponentBlur}
+         onMouseDown={handleMouseDown}
+         { ...attrs }>
+      <NxTextInput role="combobox"
+                   ref={inputRef}
+                   id={inputId}
+                   isPristine
+                   {...inputProps}
+                   className={inputClassName}
+                   value={searchText}
+                   onChange={handleFilterChange}
+                   disabled={disabled || undefined}
+                   onKeyDown={handleKeyDown}
+                   aria-autocomplete={autoComplete ? 'both' : 'list'}
+                   aria-expanded={showDropdown}
+                   aria-controls={dropdownId}
+                   aria-activedescendant={elToFocusId}
+                   aria-required={ariaRequired ?? undefined}
+                   aria-describedby={ariaDescribedBy ?? undefined}/>
+      <NxDropdownMenu id={dropdownId}
+                      role={dropdownRole}
+                      ref={dropdownRef}
+                      className={dropdownClassName}
+                      onClosing={() => {}}
+                      aria-busy={!!loading}
+                      aria-live="polite"
+                      aria-hidden={!showDropdown}>
+        <NxLoadWrapper { ...{ loading, error } } retryHandler={() => doSearch(searchText)}>
+          {
+            matches.length ? matches.map((match, i) =>
+              <button id={`${dropdownBtnId}-${i}`}
+                      role="option"
+                      aria-selected={i === focusableBtnIndex && !!elToFocusId}
+                      className= {classnames('nx-dropdown-button',
+                          { 'nx-combobox__option--visual-selected': i === focusableBtnIndex && !!elToFocusId})}
+                      tabIndex={-1}
```

Comment:
There are two ways to manage focus within compound widgets such as combobox. One way is to have a single actually-focusable element and then to manage moving the perceived focus using `aria-activedescendant`, as you have done.  The other way is a roving tabindex, where there is still only one focusable element within the component _at a time_ but which element that is changes as the user moves around.  This is how `NxSearchDropdown` does it.  Given that this code is similar to/based on `NxSearchDropdown`, I'm curious if there was a reason that you switched to the `aria-activedescendant` strategy

Reply:
In the ARIA doc of combobox, it mentioned the popup descendants are excluded from the page tab sequence.
So when navigating through the options within the listbox, the DOM Focus is maintained on the combobox and the assistive technology focus is moved within the listbox using `aria-activedescendant`.

Reply:
> In the ARIA doc of combobox, it mentioned the popup descendants are excluded from the page tab sequence.
So when navigating through the options within the listbox, the DOM Focus is maintained on the combobox and the assistive technology focus is moved within the listbox using aria-activedescendant.

I'm seeing conflicting information on that point.  The [patterns guide](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/) seems to indicate that only `aria-activedescendant` can be used and no a roving tab index.  It says:
> When a descendant of a listbox, grid, or tree popup is focused, DOM focus remains on the combobox and the combobox has [aria-activedescendant](https://w3c.github.io/aria/#aria-activedescendant) set to a value that refers to the focused element within the popup.

However, the actual [ARIA Spec](https://www.w3.org/TR/wai-aria-1.2/#combobox) makes it sound like that's optional, which to me implies that a roving tabindex is also allowed:
> If the popup element supports [aria-activedescendant](https://www.w3.org/TR/wai-aria-1.2/#aria-activedescendant), in lieu of moving focus, such keyboard mechanisms can control the value of [aria-activedescendant](https://www.w3.org/TR/wai-aria-1.2/#aria-activedescendant) on the combobox element. When a descendant of the popup element is active, authors MAY set [aria-activedescendant](https://www.w3.org/TR/wai-aria-1.2/#aria-activedescendant) on the combobox to a value that refers to the active element within the popup while focus remains on the combobox element.

Since we now have examples of both (between `NxSearchDropdown` and your current implementation here), I leave it up to you which you prefer to move forward with.

Reply:
I will keep it as an example of implementation of managing focus with `aria-activedescendant`. Thanks.